### PR TITLE
Phobia DLC - Blood and Claustrophobia

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request

**_Have you ever been afraid of being trapped in tight spaces?_** 
**_Does viewing suspicious red puddles on the floor terrify you?_** 

Then this is the PR for you!  It comes with a lifetime of paranoia guarantee.  An exclusive BOGO coder sale that is a limited time offer.  This amazing product comes with:

- **Claustrophobia Quirk** - Mobs now become terrified of being stuck inside any container and will start hyperventilating causing suffocation.  This is triggered from being inside closets, crates, coffins, cryo, DNA machines, body bags, disposals, etc. 
- **Blood Phobia** - Mobs now panic when they see any form of blood on the floor, on clothing, or blood related objects like an IV drip or bloodbag.  Recommended especially for crew that love to play medical roles.

Please redeem at your nearest /tg/ code repository while products last.

Special thanks to @TheBonded for helping me test and balance this out properly.

## Why It's Good For The Game

More RP opportunities and more "fun". 

## Changelog

:cl: timothymtorres, TheBonded
expansion: Adds Claustrophobia quirk that causes suffocation if a player enters any type of container object. (lockers, crates, machinery, disposals, etc.) 
expansion: Adds Blood Phobia that is triggered from seeing gibs, blood, bloody clothes or weapons, and blood related objects like IV drips and blood bags. 
fix: Fixed some quirks to not process if player is sleeping or unconscious.  
/:cl:

### CODE REVIEW

**Comment by timothymtorres** [comment]
I would like a maintainer to tell me if my status_effects changes will work correctly:

```if(quirk_holder.stat != CONSCIOUS || quirk_holder.has_status_effect(STATUS_EFFECT_INCAPACITATED) || quirk_holder.has_status_effect(STATUS_EFFECT_SLEEPING))```

This is the only change that was made where me and @TheBonded weren't able to test.

**Comment by EricZilla** [comment]
Another cool phobia set, very nice!
Reminds me that I meant to try and add needle phobia

**Comment by jughu** [comment]
blood phobia just means you gonna scream 24/7 :^)

**Comment by AlinaStarkova** [comment]
"cryo"

make sure they dont suffocate if cryo is on, they are supposed to be unconcious when its on afterall and as such wont hyperventilate

**Comment by timothymtorres** [comment]
If you are unconscious, sleeping, or in crit then the phobia will not trigger.

**Comment by notfrying1pans** [comment]
"Recommended especially for crew that love to play medical roles."
guns phobia as a medical main is already suffering why would you let me do this

File: code/datums/quirks/negative.dm [hunk 1]
```diff
@@ -850,3 +850,28 @@
...
+		return
+
+	if(isturf(quirk_holder.loc))
+		SEND_SIGNAL(quirk_holder, COMSIG_CLEAR_MOOD_EVENT, "claustrophobia", /datum/mood_event/claustrophobia)
+		return 
+	
+	SEND_SIGNAL(quirk_holder, COMSIG_ADD_MOOD_EVENT, "claustrophobia")
+	quirk_holder.losebreath += 0.25 // miss a breath one in four times
+	if(prob(25))
+		to_chat(quirk_holder, span_warning("You feel trapped!  Must escape... can't breath..."))
```
Comment (MrMelbert) [inline code/datums/quirks/negative.dm:877]: This `prob()` should be a `DT_PROB()` as it's processing on delta_time

**Review by MrMelbert — CHANGES_REQUESTED** [review]
One minor thing

I also wish there was a less snowflakey way of adding unique triggers per phobia instead of having to do `phobia_type == "blah"`. But oh well

File: code/datums/quirks/negative.dm (hunk 1 above)
Comment (timothymtorres) [reply code/datums/quirks/negative.dm]: Thanks for pointing this out.  Fixed with https://github.com/tgstation/tgstation/pull/63741/commits/079f3b3eaa042830634436788436ffe0cf737d1f

**Comment by timothymtorres** [comment]
> One minor thing

It's good to go now.

> I also wish there was a less snowflakey way of adding unique triggers per phobia instead of having to do `phobia_type == "blah"`. But oh well

Agreed.  I didn't like my solution but there isn't really a more optimal way without having to reloop through all the objects in a separate proc or something.

**Pull request merged**